src/main/java/org/opentripplanner/routing/core/vehicle_sharing/VehicleValidator.java

```diff
@@ -14,4 +18,12 @@ public void addFilter(VehicleFilter filter) {
     public boolean isValid(VehicleDescription vehicle) {
         return filters.stream().allMatch(f -> f.isValid(vehicle));
     }
+
+    public Set<VehicleType> getVehicleTypesAllowed() {
+        return filters.stream()
+                .filter(VehicleTypeFilter.class::isInstance)
+                .map(VehicleTypeFilter.class::cast)
+                .map(VehicleTypeFilter::getVehicleTypes)
+                .collect(() -> new HashSet<>(of(VehicleType.values())), Set::retainAll, Set::retainAll);
```

Comment:
ciekawe... daje za to okejke
streamy są mega fajne w javie, ale są takie brzydkie...

Reply:
folds in java are crazy heh